libfwupd/fwupd-security-attr.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <glib-object.h>
+
+#include "fwupd-enums.h"
+
+G_BEGIN_DECLS
+
+#define FWUPD_TYPE_SECURITY_ATTR (fwupd_security_attr_get_type ())
+G_DECLARE_DERIVABLE_TYPE (FwupdSecurityAttr, fwupd_security_attr, FWUPD, SECURITY_ATTR, GObject)
+
+struct _FwupdSecurityAttrClass
+{
+	GObjectClass			 parent_class;
+	/*< private >*/
+	void (*_fwupd_reserved1)	(void);
+	void (*_fwupd_reserved2)	(void);
+	void (*_fwupd_reserved3)	(void);
+	void (*_fwupd_reserved4)	(void);
+	void (*_fwupd_reserved5)	(void);
+	void (*_fwupd_reserved6)	(void);
+	void (*_fwupd_reserved7)	(void);
+};
+
+
+/**
+ * FwupdSecurityAttrFlags:
+ * @FWUPD_SECURITY_ATTR_FLAG_NONE:			No flags set
+ * @FWUPD_SECURITY_ATTR_FLAG_SUCCESS:			Success
+ * @FWUPD_SECURITY_ATTR_FLAG_RUNTIME_UPDATES:		Suffix `U`
+ * @FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ATTESTATION:	Suffix `A`
+ * @FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE:		Suffix `!`
+ * @FWUPD_SECURITY_ATTR_FLAG_RUNTIME_UNTRUSTED:		Suffix `?`
+ *
+ * The flags available for HSI attributes.
+ **/
+typedef enum {
+	FWUPD_SECURITY_ATTR_FLAG_NONE			= 0,
+	FWUPD_SECURITY_ATTR_FLAG_SUCCESS		= 1 << 0,
+	FWUPD_SECURITY_ATTR_FLAG_RUNTIME_UPDATES	= 1 << 8,
```

Comment:
Why are you starting at 8 here?

Reply:
I left a bit of room for "standard" flags so we could keep the RUNTIME ones together. For instance I think I need a `FWUPD_SECURITY_ATTR_FLAG_OBSOLETED` so we can run though all the obsoletes after running all the vfuncs and then easily skip. Maybe not, I've not implemented this bit yet.

Reply:
Ah OK.  The logic being there will probably be a lot more runtime ones later that can factor into stuff.

Reply:
Right.